File: onlinepayments-sdk/src/main/java/com/onlinepayments/sdk/client/android/model/api/ApiErrorItem.java
/*
 * Copyright 2020 Global Collect Services B.V
 */

package com.onlinepayments.sdk.client.android.model.api;

import java.io.Serializable;

/**
 * The error item in the error body.
 */
public class ApiErrorItem implements Serializable {

    private static final long serialVersionUID = 1983759919374923872L;

    private String errorCode;
    private String category;
    private Integer httpStatusCode;
    private String id;
    private String propertyName;
    private boolean retriable = true;

    /**
     * The error code
     *
     * @deprecated In a future release, this property will be removed. Use error code instead.
     */
    @Deprecated
    public String code = "This error does not contain a code";

    /**
     * The error message
     */
    public String message = "This error does not contain a message";

    /**
     * The error code
     */
    public String getErrorCode() { return errorCode; }

    /**
     * The category the error belongs to
     */
    public String getCategory() { return category; }

    /**
     * The HTTP status code
     */
    public Integer getHttpStatusCode() { return httpStatusCode; }

    /**
     * The error id
     */
    public String getId() { return id; }

    /**
     * The name of the property
     */
    public String getPropertyName() { return propertyName; }

    /**
     * Indicating whether the request is retriable
     */
    public boolean isRetriable() { return retriable; }
}

File: onlinepayments-sdk/src/main/kotlin/com/onlinepayments/sdk/client/android/model/api/ApiErrorItem.kt
/*
 * Do not remove or alter the notices in this preamble.
 *
 * Copyright © 2025 Worldline and/or its affiliates.
 *
 * All rights reserved. License grant and user rights and obligations according to the applicable license agreement.
 *
 * Please contact Worldline for questions regarding license and user rights.
 */
package com.onlinepayments.sdk.client.android.model.api

import java.io.Serializable

/**
 * The error item in the error body.
 */
data class ApiErrorItem(
    /** The error code returned by the server (read-only). */
    val errorCode: String? = null,

    /** The category the error belongs to (read-only). */
    val category: String? = null,

    /** The HTTP status code if available (read-only). */
    val httpStatusCode: Int? = null,

    /** The error id if available (read-only). */
    val id: String? = null,

    /** The name of the property that triggered the error if available (read-only). */
    val propertyName: String? = null,

    /** Indicates whether the request is retriable (read-only). */
    val retriable: Boolean = true,

    /**
     * A code used by older calls. Marked deprecated; new code should rely on [errorCode].
     * The default is "This error does not contain a code".
     */
    @Deprecated("In a future release, this property will be removed. Use errorCode instead.")
    var code: String = "This error does not contain a code",

    /**
     * A human-readable error message. Defaults to "This error does not contain a message".
     */
    var message: String = "This error does not contain a message"
) : Serializable {

    fun isRetriable(): Boolean = retriable

    companion object {
        @Suppress("Unused")
        private val serialVersionUID = 1983759919374923872L
    }
}
